### ISSUE TITLE
fd_install fires events for non-socket fds

### ISSUE THREAD

**Issue by simonklb** [weaveworks/tcptracer-bpf#57]
In Scope I was seeing the error message `file "/proc/20654/fd/140" is not a socket`. I noticed that the error message only showed up in the log when Scope was launched _after_ the process was started. After some digging I found that processes having an `accept()` call open when launching Scope are included in the `AddFdInstallWatcher()` call which in turn makes the `fd_install` probes fire events for those pids.

I believe the issue stem from the `fd_install` kprobe not filtering out fds that are not sockets, however, I'm new to Scope and BPF so take my investigation with a grain of salt. :)

**Comment by rade** [comment]
seems plausible. @iaguis what do you think?

**Comment by iaguis** [comment]
Yes, [we don't do any filtering](https://github.com/weaveworks/tcptracer-bpf/blob/9b0894bde621b654bed5c2e1831d295e3a0b7221/tcptracer-bpf.c#L866) on the BPF side. It seems it'd make sense doing it to avoid unnecessary processing.

**Comment by iaguis** [comment]
Note this is not trivial since we want to stay kernel-version independent and we'd need to get the offset of the field where the `i_mode` is in `struct inode`.

It probably makes more sense to not make this message an error, but a debug log in scope.

**Comment by rade** [comment]
> It probably makes more sense to not make this message an error, but a debug log in scope.

I was thinking the same. PRs welcome :)

**Comment by simonklb** [comment]
Since the purpose of this program is to track *TCP events*, letting this slide sounds like a suboptimal solution to me. Sure you can mitigate this in Scope, but it would be an odd thing to ignore if you want to see it being used by others.
